### ISSUE TITLE
MySQL: parse error with table name _

### ISSUE THREAD

**Issue by yoichi** [sqlfluff/sqlfluff#3944]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff with MySQL dialect cause parse errror with the table name _.

Where the table name _ is valid as

```
mysql> create table _ (a int);
Query OK, 0 rows affected (0.03 sec)

mysql> show create table _\G
*************************** 1. row ***************************
       Table: _
Create Table: CREATE TABLE `_` (
  `a` int DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
1 row in set (0.02 sec)

mysql> select * from _;
Empty set (0.01 sec)

mysql>
```

tested with MySQL 8.0.29.

### Expected Behaviour

sqlfluff does not cause parse error

### Observed Behaviour

sqlfluff causes parse error

### How to reproduce

```
$ cat select.sql
select * from _;
$ sqlfluff lint select.sql --dialect=mysql
== [select.sql] FAIL
L:   1 | P:   1 | L044 | Query produces an unknown number of result columns.
L:   1 | P:  10 |  PRS | Line 1, Position 10: Found unparsable section: 'from _'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Note: parse does not occur with other dialects:

```
$ sqlfluff lint select.sql --dialect=bigquery
== [select.sql] FAIL
L:   1 | P:   1 | L044 | Query produces an unknown number of result columns.
L:   1 | P:  15 | L057 | Do not use special characters in identifiers.
All Finished 📜 🎉!
```

```
$ sqlfluff lint select.sql --dialect=postgres
== [select.sql] FAIL
L:   1 | P:   1 | L044 | Query produces an unknown number of result columns.
L:   1 | P:  15 | L057 | Do not use special characters in identifiers.
All Finished 📜 🎉!
```

### Dialect

mysql

### Version

sqlfluff, version 1.3.2

### Configuration

no configuration file

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by nimish-nimishmittal** [comment]
Can i work upon this please ??
I know about this issue and have learnt this topic...

**Comment by barrywhart** [comment]
Of course, thanks! I assigned it to you.
